tsdb/head_wal.go

```diff
@@ -437,13 +437,14 @@ func (wp *walSubsetProcessor) waitUntilIdle() {
 	default:
 	}
 	wp.input <- []record.RefSample{}
+	ticker := time.NewTicker(10 * time.Microsecond)
 	for len(wp.input) != 0 {
-		time.Sleep(10 * time.Microsecond)
 		select {
 		case <-wp.output: // Allow output side to drain to avoid deadlock.
-		default:
+		case <-ticker.C:
```

Comment:
Why not using `<-time.After(10 * time.Microsecond)` ?

Reply:
I'm going to try this and benchmark it as well, results coming in soon.

Reply:
So i got these results from running each benchmark 5 times:
```
❯ benchstat new_implementation_using_ticker.txt new_implementation_using_time_after.txt
name                                                                                                     old time/op  new time/op  delta
LoadWAL/batches=10,seriesPerBatch=100,samplesPerSeries=7200,exemplarsPerSeries=0,mmappedChunkT=0-8        426ms ±14%   253ms ± 6%   -40.64%  (p=0.008 n=5+5)
LoadWAL/batches=10,seriesPerBatch=100,samplesPerSeries=7200,exemplarsPerSeries=36,mmappedChunkT=0-8       421ms ± 4%   369ms ±31%      ~     (p=0.690 n=5+5)
LoadWAL/batches=10,seriesPerBatch=100,samplesPerSeries=7200,exemplarsPerSeries=72,mmappedChunkT=0-8       401ms ± 2%   297ms ±28%   -25.85%  (p=0.008 n=5+5)
LoadWAL/batches=10,seriesPerBatch=100,samplesPerSeries=7200,exemplarsPerSeries=360,mmappedChunkT=0-8      405ms ±13%  999ms ±187%      ~     (p=0.690 n=5+5)
LoadWAL/batches=10,seriesPerBatch=10000,samplesPerSeries=50,exemplarsPerSeries=0,mmappedChunkT=0-8        575ms ±10%  1304ms ±44%  +126.88%  (p=0.008 n=5+5)
LoadWAL/batches=10,seriesPerBatch=10000,samplesPerSeries=50,exemplarsPerSeries=2,mmappedChunkT=0-8        575ms ± 3%  1211ms ±20%  +110.53%  (p=0.008 n=5+5)
LoadWAL/batches=10,seriesPerBatch=1000,samplesPerSeries=480,exemplarsPerSeries=0,mmappedChunkT=0-8        343ms ±83%   176ms ± 7%   -48.68%  (p=0.016 n=5+5)
LoadWAL/batches=10,seriesPerBatch=1000,samplesPerSeries=480,exemplarsPerSeries=2,mmappedChunkT=0-8        1.03s ± 3%   0.18s ± 5%   -82.32%  (p=0.016 n=4+5)
LoadWAL/batches=10,seriesPerBatch=1000,samplesPerSeries=480,exemplarsPerSeries=5,mmappedChunkT=0-8        237ms ±22%   274ms ±33%      ~     (p=0.548 n=5+5)
LoadWAL/batches=10,seriesPerBatch=1000,samplesPerSeries=480,exemplarsPerSeries=24,mmappedChunkT=0-8       245ms ± 3%   364ms ± 7%   +48.55%  (p=0.008 n=5+5)
LoadWAL/batches=100,seriesPerBatch=1000,samplesPerSeries=480,exemplarsPerSeries=0,mmappedChunkT=3800-8    2.65s ±50%   3.10s ± 6%      ~     (p=0.151 n=5+5)
LoadWAL/batches=100,seriesPerBatch=1000,samplesPerSeries=480,exemplarsPerSeries=2,mmappedChunkT=3800-8    3.57s ±15%   3.40s ± 4%      ~     (p=0.548 n=5+5)
LoadWAL/batches=100,seriesPerBatch=1000,samplesPerSeries=480,exemplarsPerSeries=5,mmappedChunkT=3800-8    4.67s ±34%   3.84s ±17%      ~     (p=0.690 n=5+5)
LoadWAL/batches=100,seriesPerBatch=1000,samplesPerSeries=480,exemplarsPerSeries=24,mmappedChunkT=3800-8   3.19s ±20%   4.00s ±18%   +25.32%  (p=0.032 n=5+5)
```

There isn't really much difference. I'll go with the time.After one for simplicity :+1: 

I'll also update the comment above with the new benchmark results.